### PR TITLE
Get rid of ActiveSupport dependency

### PR DESCRIPTION
**Pull request by barthez**
Hey,

I thought it would be nice to simplify the gem and remove unnecessary dependency on active support. There was only 1 usage of `Class#subclasses` method that could be easily replaced with `inherited` hook giving the same end result - the first class that inherited from `Karafka::BaseWorker`.

### CODE REVIEW

**Comment by mensfeld** [comment]
Hey. First of all, thank you for the PR. Don't worry about failing specs. Coditsu does not yet support forks :( (I'm working on it though).

I'm currently in Briston but @nijikon will review this code initially and I will get back to that in couple days.

This change is something I've wanted to add for a while now as gradually we're moving out of activesupport completely.

File: spec/lib/karafka/workers/builder_spec.rb [hunk 1]
```diff
@@ -80,10 +82,10 @@ class SuperSadConsumer
         end
 
         it 'expect to build it' do
-          expect(builder.build.to_s).to eq 'SuperSadWorker'
+          is_expected.to be < base_worker
```
Comment (nijikon) [inline spec/lib/karafka/workers/builder_spec.rb]: Please use separate `it` block for each `expect` that you are testing.

**Comment by barthez** [comment]
Thank @nijikon for the review. Addressed all the comments.

**Comment by mensfeld** [comment]
LGTM. Merging. I will apply quality improvements for this one today. Thanks!

**Pull request merged**